src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,22 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- has-invalid-json? [{:keys [name type]}]
+  (when (= :json type)
+    (try
+      ;; parse, but ignore the result
+      (json/parse-string (get-value-of-type :string name))
+      false
+      (catch JsonParseException _ true)
+      ;; something else went wrong, and it could be masking an issue with the json value
+      ;; we don't want to return false positives, but indicate an issue by returning a different falsey value
+      (catch Exception _ nil))))
+
+(defn validate-json-settings!
+  "Throws an exception if there are any settings with invalid JSON configuration.
+   Note that this will only validate settings whose defsetting forms have already been evaluated."
+  []
+  (when-let [invalid-settings (seq (map key (filter (comp has-invalid-json? val) @registered-settings)))]
+    (throw (ex-info (tru "Invalid JSON configuration for settings: {0}" invalid-settings)
```

Comment:
`tru` means the string will be translated into the language of "the user" meaning an API user, which is only used when the message is passed to the frontend for display. See [`metabase.util.i18n/translate-user-locale`](https://github.com/metabase/metabase/blob/cd7a8c71821dae559e83fc92ba3173e7b8d6fa94/src/metabase/util/i18n.clj#L78) to understand what's going on. We use `trs` for this kind of message that is meant to be translated into the language of the metabase instance, see [this](https://github.com/metabase/metabase/blob/cd7a8c71821dae559e83fc92ba3173e7b8d6fa94/enterprise/backend/src/metabase_enterprise/advanced_config/file.clj#L211) example.
```suggestion
    (throw (ex-info (trs "Invalid JSON configuration for settings: {0}" invalid-settings)
```

Reply:
As for this error message, I think we should display the exception message from the `JsonParseException`, because it gives useful information about what is invalid about the JSON string. And I'd say it's okay to raise one exception at a time to simplify the display and the code. The operator is likely to only have one setting in the incorrect format at a time anyway.